sympy/sets/fancysets.py

```diff
@@ -903,7 +903,7 @@ def _union(self, other):
             elif self.polar and other.polar:
                 return ComplexRegion(Union(self.sets, other.sets), polar=True)
 
-        if other.is_subset(S.Reals):
+        if self == S.Complexes:
```

Comment:
No, I don't think so. Please note that `S.Complexes` is a special case singleton of `ComplexRegion`, i.e. it's just `ComplexRegion(Interval(-oo, oo)*Interval(-oo, oo))`. Hence it should inherit the properties of `ComplexRegion(Interval(-oo, oo)*Interval(-oo, oo))`, rather than defining it's own methods.